src/xwrap.rs

```diff
@@ -125,6 +146,29 @@ impl Display {
             })
         }
     }
+
+    pub fn get_cursor_position(&self, mut root_win: xlib::Window) -> util::Point {
+        let mut x = 0;
+        let mut y = 0;
+
+        let (mut win_x, mut win_y, mut mask) = (0, 0, 0);
+
+        unsafe {
+            xlib::XQueryPointer(
+                self.handle,
+                root_win,
+                &mut root_win,
+                &mut root_win,
+                &mut x,
+                &mut y,
+                &mut win_x,
+                &mut win_y,
+                &mut mask,
+            );
+        }
+
+        util::Point { x, y }
+    }
 }
```

Comment:
So according to the documentation, XQueryPointer *is* fallible. https://tronche.com/gui/x/xlib/window-information/XQueryPointer.html

The first failure mode is a BadWindow error if you pass in an invalid window. For our use that would be a programming error so you can just make that a panic. (actually xlib will very likely do that itself so maybe it's not necessary, try passing it some random integer and see what happens)

It looks like the other failure case is related to having multiple X screens (not the same as multiple monitors). That is a very uncommon configuration that we shouldn't bother supporting. Regardless, you should re-introduce the Option and the error message for when XQueryPointer returns false.

Lastly, the root_win argument should just be called window, and it shouldn't be mutable. This is a big C-ism that we don't have to deal with in Rust. If you want to use the values XQueryPointer puts in there, you should put them in the return struct. It's unnecessary in this case though, so you should just throw root_return and child_return away.

I do wonder, what happens if you pass 0 as the window argument? If it works, then we should still get the cursor position relative to the root window and you can get rid of the window argument. Otherwise, win_x and win_y are the values you should be returning, as they are relative to the window you're passing in.

Reply:
Ok, I'll make sure that I get that done soon, unfortunately this is the first project I worked on with rust so I'll have to take a closer look to the error handing part.